client/src/pages/Search/AddressSearch.js

```diff
@@ -2,8 +2,23 @@ import { IconButton, InputAdornment, TextField } from "@mui/material";
 import { usePlacesWidget } from "react-google-autocomplete";
 import { HUNTER_COLLEGE_ADDRESS } from "./constants";
 import ClearIcon from "@mui/icons-material/Clear";
+import getUserCoordinates from "../../utils/getUserCoordinates";
+import axios from "axios";
+import MyLocationIcon from "@mui/icons-material/MyLocation";
+import { useState } from "react";
 
 export default function AddressSearch({ updateFields }) {
+  const [address, setAddress] = useState(HUNTER_COLLEGE_ADDRESS);
+
+  async function handleClick() {
+    const coordinates = await getUserCoordinates();
+    const latitude = coordinates.coordinates.latitude;
+    const longitude = coordinates.coordinates.longitude;
+    const reversegeolocate = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.REACT_APP_GOOGLE_API_KEY}`;
+    axios.get(reversegeolocate).then(function (response) {
+      setAddress(response.data.results[0].formatted_address);
```

Comment:
use googlePlacesRef instead (2 lines under)

Read the docs to see how to use refs
https://react.dev/reference/react/useRef